.github/workflows/integration_tests.yml

```diff
@@ -771,56 +625,132 @@ jobs:
         with:
-          name: test-report
+          name: test-report-json
+          path: ${{ env.TEST_REPORTS_PATH }}
+
+      - name: Build multi-platform test report
+        uses: actions/github-script@v5.0.0
+        env:
+          TEST_REPORTS_PATH: ${{ env.TEST_REPORTS_PATH }}
+          FINAL_TEST_REPORTS_FOLDER: ${{ env.FINAL_TEST_REPORTS_FOLDER }}
+          FINAL_TEST_REPORT_FILEPATH_JSON: ${{ env.FINAL_TEST_REPORT_FILEPATH_JSON }}
+        with:
+          script: |
+            ndJsonParser = require('ndjson-parse');
+            fs = require('fs');
+            
+            const {TEST_REPORTS_PATH, FINAL_TEST_REPORT_FILEPATH_JSON, FINAL_TEST_REPORTS_FOLDER} = process.env
+            const jsonReportData = [];
+            
+            const fileList = fs.readdirSync(TEST_REPORTS_PATH);
+            fileList.forEach(fileName => {1
+                console.log(`Reading file: ${fileName}`);
+                const platformReportFile = fs.readFileSync(TEST_REPORTS_PATH + fileName, {encoding:'utf8', flag:'r'});
+                const platformTestReportRaw = ndJsonParser(platformReportFile);
+                
+                platformTestReportRaw.forEach(platformTestResult => {
+                    if(platformTestResult.test.includes("/")) {
+                        jsonReportData.push(platformTestResult);
+                    }
+                });
+            });
+            
+            const testResultMap = {};
+            
+            jsonReportData.forEach(testResult => {
+                const trimmedTestResult = { ...testResult, test: testResult.test.split("/")[1] }
+                
+                // If this is a new testcase the result list needs to be created first
+                if (!(trimmedTestResult.test in testResultMap)) {
+                    testResultMap[trimmedTestResult.test] = [];
+                }
+                
+                // Add to result list
+                testResultMap[trimmedTestResult.test].push(trimmedTestResult)
+            });
+            
+            const finalTestReport = [];
+            const allPlatforms = [];
+            
+            // Create properly formatted final test report
+            Object.entries(testResultMap).forEach(([test, testResults]) => {
+                const finalResult = {};
+                finalResult.test = test;
+                testResults.forEach(result => {
+                    const {test, ...results} = result;
+                    Object.assign(finalResult, results);
+                    
+                    const currentPlatform = Object.keys(results)[0];
+                    if(!allPlatforms.includes(currentPlatform)) {
+                        allPlatforms.push(currentPlatform);
+                    }
+                });
+                finalTestReport.push(finalResult);
+            });
+            
+            // Fill in platforms where a test was not executed
+            finalTestReport.forEach(testResult => {
```

Comment:
Add "skip" test result entries to the testcases where not all platforms were configured to run those testcases